crates/mun_gc/src/lib.rs

```diff
@@ -0,0 +1,89 @@
+mod handle;
+mod mark_sweep;
+mod root_handle;
+
+pub use handle::{GCPtr, HasIndirectionPtr, RawGCPtr};
+pub use mark_sweep::MarkSweep;
+pub use root_handle::GCRootHandle;
+
+/// Contains stats about the current state of a GC implementation
+#[derive(Debug, Clone, Default)]
+pub struct Stats {
+    pub allocated_memory: usize,
+}
+
+/// A trait used by the GC to identify an object.
+pub trait Type: Send + Sync {
+    type Trace: Iterator<Item = GCPtr>;
+
+    /// Returns the size in bytes of an object of this type.
+    fn size(&self) -> usize;
+
+    /// Returns the alignment of a type
+    fn alignment(&self) -> usize;
+
+    /// Returns an iterator to iterate over all GC objects that are referenced by the given object.
+    fn trace(&self, obj: GCPtr) -> Self::Trace;
+}
+
+/// An object that can be used to allocate and collect memory.
+pub trait GCRuntime<T: Type>: Send + Sync {
```

Comment:
In Rust naming conventions, acronyms are seem as [one word](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#general-conventions-[rfc-#430]). So the preferred way of naming is `GcRuntime` and `GcPtr`.